packages/eas-cli/src/commands/deploy/index.ts

```diff
@@ -299,6 +300,25 @@ export default class WorkerDeploy extends EasCommand {
       progress.succeed('Created deployment');
     } catch (error: any) {
       progress.fail('Failed to create deployment');
+
+      if (
+        flags.isProduction &&
+        error instanceof CombinedError &&
+        error.graphQLErrors.some(err => {
+          return (
+            err.extensions?.errorCode === 'UNAUTHORIZED_ERROR' &&
+            err.message.includes('AppDevDomainNameEntity') &&
+            err.message.includes('CREATE')
+          );
+        })
+      ) {
+        throw new Error(
+          `You have specified the new deployment should be a production deployment, but a production domain has not been set up yet for this app.\n\nRun ${chalk.bold(
+            'eas deploy --prod'
+          )} as an app admin on your machine to set up a production domain and then try again.`
```

Comment:
Another way to do this would be to promote an existing deployment to production on the dashboard https://expo.dev/accounts/[account]/projects/[project]/hosting/deployments?view=deployments

```suggestion
          )} as an app admin on your machine or promote an existing deployment to production on the Hosting Dashbaord to set up a production domain and then try again.`
```